addons/common/fnc_runQTE.sqf

```diff
@@ -0,0 +1,163 @@
+#include "script_component.hpp"
+#include "\a3\ui_f\hpp\defineDIKCodes.inc"
+/*
+	 * Author: john681611
+	 * Runs a Quick time Event.
+	 *
+	 * Arguments:
+	 * 0: Object QTE is attached to <OBJECT>
+	 * 1: Arguments, passed to condition, fail and finish <ARRAY>
+	     * 2: On Display: Code callback on displayable event. <CODE, STRING>
+	 * 2: On Finish: Code called or STRING raised as event. <CODE, STRING>
+	 * 3: On Failure: Code called or STRING raised as event. <CODE, STRING>
+	 * 4: QTE Seqence <ARRAY>
+	 * 5: max interaction distance from attached object <NUMBER> (default: 10) 
+	 * 6: timout <NUMBER> (default: 30) 
+	 *
+	 * Return Value:
+	 * None
+	 *
+	 * Example:
+	 * [car, [], {
+		hint "Finished!"
+	}, {
+		hint "Failure!"
+	}, ["↑", "↓", "→", "←"]] call ace_common_fnc_runQTE
+	 *
+	 * Public: Yes
+ */
+
+params ["_object", "_args", "_onDisplay", "_onFinish", "_onFail", "_qte_seqence", ["_max_distance", 10], ["_timeout", 30]];
+
+if (!GVAR(settingsInitFinished)) exitWith {
+	// only run this after the settings are initialized
+	GVAR(runAtSettingsInitialized) pushBack [FUNC(runQTE), _this];
+};
+
+if (GVAR(QTERunning)) exitWith {
+	TRACE_1("QTE already running qeueing up", GVAR(QTERunning));
+	[{
+		!GVAR(QTERunning)
+	}, {
+		_this call FUNC(runQTE);
+	}, _this] call CBA_fnc_waitUntilAndExecute;
+};
+
+private _display = findDisplay 46;
+
+if (isNull _display) exitWith {
+	TRACE_1("Waiting for main display to be ready", isNull (_display));
+	[{
+		!isNull (findDisplay 46)
+	}, {
+		_this call FUNC(runQTE);
+	}, _this] call CBA_fnc_waitUntilAndExecute;
```

Comment:
Imo it should be up to the user to check whenever it's ready. This should only return `false` or an error code that it couldn't run. The removal of these `waitUntilAndExecute` would make the `_onDisplay` event redundant.

~~The display on which the QTE is drawn should be an argument, so that people can run it wherever they want to (e.g. Zeus interface).~~ Because of comment below, this is no longer relevant.